src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -288,7 +288,7 @@ def schema(self, schema: StructType) -> "DataFrameReader":
         return self
 
     def with_metadata(
-        self, *metadata_cols: Iterable[_MetadataColumn]
+        self, *metadata_cols: Iterable[ColumnOrName]
```

Comment:
Do you want to use [typing.Literal](https://docs.python.org/3/library/typing.html#typing.Literal) to limit the input types to those constants?

Reply:
Literal only allows `None`, literal values - `int`, `str`, `bytes`, `bool`, or enum values. I would have to create enums for Metadatacolumn which would be more work than necessary. I am happy with `ColumnOrName`.

Reply:
OK, forget it then.